packages/shared/src/hooks/useTimedAnimation.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { isNullOrUndefined } from '../lib/func';
+
+interface UseTimedAnimation {
+  timer: number;
+  isAnimating: boolean;
+  endAnimation: () => void;
+}
+
+interface UseTimedAnimationProps {
+  autoEndAnimation: boolean;
+  animationDuration: number;
+  onAnimationEnd?: () => void;
+  onAnimationStop?: () => void;
+}
+
+const TEMPORARY_ID = 1;
```

Comment:
Why do we need this instead of using null?

Reply:
Ok I see now. In this case I'd just rename this constant as it's not clear that this is its purpose